plugins/pulp_rpm/plugins/importers/yum/config_validate.py

```diff
@@ -25,3 +46,69 @@ def validate(config):
             msg += failure_message + '\n'
         msg = msg.rstrip()  # remove the trailing \n
         return False, msg
+
+
+def _validate_allow_keys(repo, config, error_messages):
```

Comment:
@mhrivnak, Actually I'm not proposing a new responsibility, I believe we
both did. You and I had about an in-person conversation early on in the
development of this feature which resulted in [this plan](
https://pulp.plan.io/issues/1991#note-10). Is there something different now
that the plan we agreed upon is not sufficient?

On Tue, Jul 26, 2016 at 10:26 AM, Michael Hrivnak <notifications@github.com>
wrote:

> In plugins/pulp_rpm/plugins/importers/yum/config_validate.py
> <https://github.com/pulp/pulp_rpm/pull/938#discussion_r72259385>:
>
> > @@ -25,3 +46,69 @@ def validate(config):
> >              msg += failure_message + '\n'
> >          msg = msg.rstrip()  # remove the trailing \n
> >          return False, msg
> > +
> > +
> > +def _validate_allow_keys(repo, config, error_messages):
>
> I think we should start by letting the user manage the list of keys,
> including the ability to remove them, and let the user decide if/when it is
> appropriate or useful to remove a key. They are free to be very strict
> about never removing keys.
>
> @bmbouter <https://github.com/bmbouter> you are thinking ahead about pulp
> taking responsibility for ongoing re-validation of content already in the
> repo, which I don't think we need to tackle at this time. To make an
> analogy, I think we should be a bouncer at the door to a repo. The bouncer
> has rules about who can get in. At any time, the owner can tell the bouncer
> to change the rules, or make exceptions for specific people. But the
> bouncer is not required to re-evaluate everyone who is already inside when
> the rules change, or when exceptions happen.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pulp/pulp_rpm/pull/938/files/6778b11542e41648d4cb5c3947c52ef6e11b50df#r72259385>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALAX6lcRHNJFZh1Gt_0aHD-B0gXIFT9ks5qZhj5gaJpZM4JQ1px>
> .
>



-- 
Brian Bouterse